File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> m;
        
        for(int i=0;i<nums.size();i++){
            int val=target-nums[i];
            if(m.find(val)!=m.end()){
                return{i,m[val]};
            }
             m[nums[i]]=i;
        }
        return{-1,-1};
    }
};

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        int[] answer = new int[2];

        HashMap<Integer, Integer> umap = new HashMap<>();

        int sum = 0;
        for (int i = 0; i < n; i++) {
            sum = nums[i];

            int rem = target-sum;

            if (umap.containsKey(rem)) {

                answer[0] = i;
                answer[1] = umap.get(rem);
                return answer;
            }
            umap.put(nums[i], i);
        }

        return answer;

    }
}